### PR TITLE
feat: add `ReadableStreamDefaultReader::Constructor`

### PR DESCRIPTION
**Pull request by pewsheen**

### CODE REVIEW

File: components/script/dom/readablestreamdefaultreader.rs [hunk 1]
```diff
@@ -41,28 +45,81 @@ impl ReadableStreamDefaultReader {
         proto: Option<SafeHandleObject>,
         stream: DomRoot<ReadableStream>,
     ) -> Fallible<DomRoot<Self>> {
-        todo!()
+        Self::new_with_proto(global, proto, stream)
```
Comment (wusyong) [inline components/script/dom/readablestreamdefaultreader.rs:48]: Can you provide doc comment that link to the spec about this constructor?

File: components/script/dom/readablestreamdefaultreader.rs [hunk 2]
```diff
@@ -41,28 +45,81 @@ impl ReadableStreamDefaultReader {
...
     }
 
-    fn new_inherited() -> ReadableStreamDefaultReader {
-        ReadableStreamDefaultReader {
-            reflector_: Reflector::new(),
-            read_requests: DomRefCell::new(Vec::new()),
-        }
+    pub fn new(global: &GlobalScope, stream: DomRoot<ReadableStream>) -> Fallible<DomRoot<Self>> {
```
Comment (wusyong) [inline components/script/dom/readablestreamdefaultreader.rs:52]: Do we still need this?

File: components/script/dom/readablestreamdefaultreader.rs [hunk 3]
```diff
@@ -41,28 +45,81 @@ impl ReadableStreamDefaultReader {
         proto: Option<SafeHandleObject>,
         stream: DomRoot<ReadableStream>,
     ) -> Fallible<DomRoot<Self>> {
-        todo!()
```
Comment (wusyong) [inline components/script/dom/readablestreamdefaultreader.rs:44]: Can we add a doc comment to this method that link the spec of this constructor?

File: components/script/dom/readablestreamdefaultreader.rs (hunk 2 above)
Comment (pewsheen) [reply components/script/dom/readablestreamdefaultreader.rs:52]: I'm not sure about the pattern here. But most other DOM types have `new` fn  and call `new_with_proto` with `proto: None`

File: components/script/dom/readablestreamdefaultreader.rs (hunk 3 above)
Comment (pewsheen) [reply components/script/dom/readablestreamdefaultreader.rs:44]: added

File: components/script/dom/readablestreamdefaultreader.rs (hunk 2 above)
Comment (wusyong) [reply components/script/dom/readablestreamdefaultreader.rs:52]: Then `new_with_proto` below should be called `new` instead.

**Review by wusyong — APPROVED** [review]

**Pull request merged**